### DISCUSSION TITLE
terminator update

### DISCUSSION THREAD

**Discussion by balrog-nona** [OpenIndiana/oi-userland#6373] [Q&A]
Hi,
I'm trying to update terminator https://github.com/gnome-terminator/terminator/releases/tag/v2.1.0 and when I try to install published packages, I get this message:
```pkg install: No matching version of terminal/terminator can be installed:
  Reject:  pkg://userland/terminal/terminator@2.1.0-2020.0.1.0
  Reason:  This version is excluded by installed incorporation consolidation/userland/userland-incorporation@0.5.11-2020.0.1.13723
No matching version of terminal/terminator-35 can be installed:
  Reject:  pkg://userland/terminal/terminator-35@2.1.0-2020.0.1.0
  Reason:  This version is excluded by installed incorporation consolidation/userland/userland-incorporation@0.5.11-2020.0.1.13723
```
I don't understand what the message is telling me.
Thanks for any suggestions.

**Comment by AndWac** [comment]
Am 05.01.21 um 13:32 schrieb Nona Hansel:
>
> Hi,
> I'm trying to update terminator
> https://github.com/gnome-terminator/terminator/releases/tag/v2.1.0
> <https://github.com/gnome-terminator/terminator/releases/tag/v2.1.0>
> and when I try to install published packages, I get this message:
>
> |Reject: pkg://userland/terminal/terminator@2.1.0-2020.0.1.0 Reason:
> This version is excluded by installed incorporation
> consolidation/userland/userland-incorporation@0.5.11-2020.0.1.13723 No
> matching version of terminal/terminator-35 can be installed: Reject:
> pkg://userland/terminal/terminator-35@2.1.0-2020.0.1.0 Reason: This
> version is excluded by installed incorporation
> consolidation/userland/userland-incorporation@0.5.11-2020.0.1.13723 |
>
> I don't understand what the message is telling me.
> Thanks for any suggestions.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/discussions/6373>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADET4N6AY64KQQOJADUKUTDSYMBFXANCNFSM4VVA6CPQ>.
>
userland-incorporation is there to fix the versions of all userland
packages (similar package is osnet-incorporation for the base os).
Such situations usually happen if your userland-incorporation is stuck
at an old version.
You'll need to find out what the latest version of it is an then
manually try to intall that version

pkg install userland-incorporation@0.5.11-2020.0.1.13724
This won't work but will tell you what packages prohibit this step.


To circumvent these kind of situations I typically do the following when
I test new packages:
I usually update to the latest:
pkg update -v
pfexec init 6

Then I create a "test" BE, activate it and boot into it:
pfexec beadm create test
pfexec beadm activate test
pfexec init 6

After that I uninstall the packages "entire" and "userland-incorporation":
pfexec pkg uninstall entire userland-incorporation

This "frees" the BE from all userland version fixations.
When done I simply activate the latest non-test BE, boot into it and
remove all test-* BE's.

Regards,
Andreas

**Comment by AndWac** [comment]
Am 05.01.21 um 13:32 schrieb Nona Hansel:
>
> Hi,
> I'm trying to update terminator
> https://github.com/gnome-terminator/terminator/releases/tag/v2.1.0
> <https://github.com/gnome-terminator/terminator/releases/tag/v2.1.0>
> and when I try to install published packages, I get this message:
>
> |Reject: pkg://userland/terminal/terminator@2.1.0-2020.0.1.0 Reason:
> This version is excluded by installed incorporation
> consolidation/userland/userland-incorporation@0.5.11-2020.0.1.13723 No
> matching version of terminal/terminator-35 can be installed: Reject:
> pkg://userland/terminal/terminator-35@2.1.0-2020.0.1.0 Reason: This
> version is excluded by installed incorporation
> consolidation/userland/userland-incorporation@0.5.11-2020.0.1.13723 |
>
> I don't understand what the message is telling me.
> Thanks for any suggestions.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/discussions/6373>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADET4N6AY64KQQOJADUKUTDSYMBFXANCNFSM4VVA6CPQ>.
>
After re-reading your message I think that you try to install the new
terminator version. That won't work as long as the packages
entire and userland-incorporation fixate all package versions. You'll
need to remove them, or even better, follow the process I described
in my first answer.

Andreas

**Reply by balrog-nona** [reply]
Thank you for your suggestions. Since I'm not so familiar with switching between BE, I removed the the packages manually and it fixed the problem for me.

**Comment by AndWac** [comment]
Am 05.01.21 um 22:31 schrieb Nona Hansel:
>
> Thank you for your suggestions. Since I'm not so familiar with
> switching between BE, I removed the the packages manually and it fixed
> the problem for me.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/discussions/6373#discussioncomment-263473>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADET4N4K3GZO66CXJHKCM2LSYOAKLANCNFSM4VVA6CPQ>.
>
Switching BE's is quite easy and a major feature of OI:
Simply enter
pfexec beadm activate <BE name>

activates the BE named <BE name> and makes it the BE when you reboot. It
manages everything in / and /var.


With
pfexec create <BE name>

you create a snapshot copy of your currently active BE and give it the
name <BE name>.


With
pfexec destroy <BE-name>

you can destroy the BE <BE name>. You can do that if you don't need it
anymore and it's neither active nor booted.

BE's are quite a nice thing as with them you don't need to worry about
updates or changes you make on your system
as long as you have another work BE. pkg update determines itself
whether it needs to create a new BE. But this can
also steered with appropriate options (look at man pkg). But always
remember that it only manages / and /var.
That is, not your user data. To protect your user data you should either
make manual snapshots or automatic snapshots
by means of timeslider. And, of course, backups :)

Andreas

**Comment by balrog-nona** [comment]
Thank you for detailed instructions. Knowing now that it's safer and special feature, I'll definitely check it out!
